File: sol/128.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        res = 0
        s = set(nums)
        while len(s) != 0:
            l = 1
            x = s.pop()
            y = x
            while y + 1 in s:
                l += 1
                y += 1
                s.remove(y)
            y = x
            while y - 1 in s:
                l += 1
                y -= 1
                s.remove(y)
            res = max(res, l)
        return res

File: sol/128/128.cpp
#include <unordered_set>

struct Solution {
    int longestConsecutive(vector<int>& nums) {
        int res = 0;
        unordered_set<int> s(nums.begin(), nums.end());
        while (!s.empty()){
            int l = 1, x = *s.begin();
            s.erase(s.begin());
            int y = x;
            while (y < 2147483647 && s.contains(y + 1)){
                ++y;
                ++l;
                s.erase(y);
            }
            y = x;
            while (y > -2147483648 && s.contains(y - 1)){
                --y;
                ++l;
                s.erase(y);
            }
            res = max(res, l);
        }
        return res;
    }
};
